CefSharp.Wpf/IME/NativeIME.cs

```diff
@@ -0,0 +1,199 @@
+        public const int WM_IME_SELECT = 0x285;
+        public const int WM_IME_CHAR = 0x286;
+        public const int WM_IME_REQUEST = 0x0288;
+        public const int WM_IME_KEYDOWN = 0x290;
+        public const int WM_IME_KEYUP = 0x291;
+        public const int WM_SYSCHAR = 0x0106;
+
+
+        internal const uint GCS_RESULTSTR = 0x0800;
+        internal const uint GCS_COMPSTR = 0x0008;
+        internal const uint GCS_COMPATTR = 0x0010;
+        internal const uint GCS_CURSORPOS = 0x0080;
+        internal const uint GCS_COMPCLAUSE = 0x0020;
+
+        internal const uint CPS_CANCEL = 0x0004;
+        internal const uint CS_NOMOVECARET = 0x4000;
+        internal const uint NI_COMPOSITIONSTR = 0x0015;
+
+        internal const uint ATTR_TARGET_CONVERTED = 0x01;
+        internal const uint ATTR_TARGET_NOTCONVERTED = 0x03;
+
+        internal const uint ISC_SHOWUICOMPOSITIONWINDOW = 0x80000000;
+
+        internal const uint CFS_DEFAULT = 0x0000;
+        internal const uint CFS_RECT = 0x0001;
+        internal const uint CFS_POINT = 0x0002;
+        internal const uint CFS_FORCE_POSITION = 0x0020;
+        internal const uint CFS_CANDIDATEPOS = 0x0040;
+        internal const uint CFS_EXCLUDE = 0x0080;
+
+        internal const uint LANG_JAPANESE = 0x11;
+        internal const uint LANG_CHINESE = 0x04;
+        internal const uint LANG_KOREAN = 0x12;
+
+
+        [StructLayout(LayoutKind.Sequential)]
+        internal struct TagPoint
+        {
+            public int X;
+            public int Y;
+        }
+
+        [StructLayout(LayoutKind.Sequential)]
+        internal struct TagRect
+        {
+            public int Left;
+            public int Top;
+            public int Right;
+            public int Bottom;
+        }
+
+        [StructLayout(LayoutKind.Sequential)]
+        internal struct TagCompositionForm
+        {
+            public uint DwStyle;
+            public TagPoint PtCurrentPos;
+            public TagRect RcArea;
+        }
+
+        [StructLayout(LayoutKind.Sequential)]
+        internal struct TagCandidateForm
+        {
+            public uint DwIndex;
+            public uint DwStyle;
+            public TagPoint PtCurrentPos;
+            public TagRect RcArea;
+        }
+
+        [StructLayout(LayoutKind.Sequential)]
+        public struct POINT
+        {
+            public int X;
+            public int Y;
+
+            public POINT(int x, int y)
+            {
+                X = x;
+                Y = y;
+            }
+        }
+
+        [StructLayout(LayoutKind.Sequential)]
+        public struct RECT
+        {
+            public int Left, Top, Right, Bottom;
+
+            public RECT(int left, int top, int right, int bottom)
+            {
+                Left = left;
+                Top = top;
+                Right = right;
+                Bottom = bottom;
+            }
+        }
+
+        [StructLayout(LayoutKind.Sequential)]
+        public struct CANDIDATEFORM
```

Comment:
Could this be merged with `tagCandidateForm`?

Reply:
Used CANDIDATEFORM instead of tagCandidateForm.